### PR TITLE
Parallel arkdb, with examples

### PR DESCRIPTION
**Pull request by 1beb**
This is a candidate example for writing windowed database out to files in parallel. 

Backgrounder: 

Currently, the only way to run in parallel is to run multiple tables at a time. But what if you have an exceptionally large table? This pull request includes a new function `window_parallel` that allows you to run a large table in parallel. 

Key points: 

1. Database connections must be created "within" the parallel sessions, so instead of passing a database object, one must pass a function that creates a database object. 
2.  Naming of parquet files was sequential in nature, but there's an edge case where the same number could be written by a number of processes. Naming has been changed to use tmpfile() format.
3. This sample uses `future.apply`, however, any parallel function could be substituted (furrr, etc). 

TODO: 

- [x] Update documentation for `arkdb`
- [x] Update tests
- [x] Split tests into separate files allowing for more focused `devtools::test_file()` usage
- [x] Strip message(nrow(data))
- [x] Lints 

Refs #21 because it supports parallelization at a different level.

### CODE REVIEW

**Comment by 1beb** [comment]
@cboettig This is ready for review.

File: README.Rmd [hunk 1]
```diff
@@ -125,6 +125,35 @@ mins_to_hours <- function(data) {
...
+
+future_lapply(vector_of_tables, function(x) ark(db, dir, lines, tables = x))
+
+# Strategy 2
+library(arkdb)
+library(future.apply)
+
+plan(multisession)
+
+ark(db, dir, lines = 50000, tables = "flights", method = "window-parallel")
```
Comment (cboettig) [inline README.Rmd:147]: Love the examples here. Just to be clear, does `method = "window-parallel"` or  imply we are writing to (chunked) parquet files?

Comment (1beb) [reply README.Rmd:147]: Yes, this is a good point. I didn't setup the other methods to accept a filename argument. SHould probably setup a stop condition here. I'll update.

Comment (1beb) [reply README.Rmd]: Resolving this. Added a stop condition + test for stop.

**Comment by 1beb** [comment]
@cboettig ready for round 2

**Comment by cboettig** [comment]
Looks good!  My only other thought is maybe to mention in the README as well that `ark` is automatically selecting `streamable_parquet()`; otherwise the reader might assume from the syntax that it is still writing out with the usual default.

**Comment by 1beb** [comment]
I actually went the other way. Instead of making the decision for someone, I chose to stop() if they used window-parallel without straeamable_parquet. I'm not sure what the best decision is here. I don't think I'd want to get parquet if I was expecting the default. Can't guarantee people will pay attention to that detail in a readme, but they will attempt to rectify a stop. Maybe?

**Comment by cboettig** [comment]
That makes sense to me.  

Only place I think may still be confusing, at least to me, is in "Strategy 1", it looks like this is not using `streamable_parquet()`? https://github.com/ropensci/arkdb/pull/48/files#diff-72778b58969c8ca8268402860b0e003e3d213a26c812bc9f9b928395c284c99fR139

File: README.Rmd [hunk 2]
```diff
@@ -125,6 +125,53 @@ mins_to_hours <- function(data) {
...
+There are two strategies for using `ark` in parallel. One is to loop over the tables, re-using the ark function per table in parallel. The other, introduced in 0.0.15, is to use the "window-parallel" method. This is particularly useful if your tables are very large and can speed up the process significantly. 
+
+```{r, eval = FALSE}
+# Strategy 1: Parallel over tables
+library(arkdb)
+library(future.apply)
+
+plan(multisession)
+
+future_lapply(vector_of_tables, function(x) ark(db, dir, lines, tables = x))
```
Comment (cboettig) [inline README.Rmd:139]: this one seems to use the default `streamable_table`, but the document doesn't mention that these other strategies are also changing the choice of output format.

Comment (1beb) [reply README.Rmd:142]: I made some updates to make this more explicit, and clear.

**Comment by cboettig** [comment]
excellent, this looks good to me!

**Pull request merged**